collector/generic_collector.go

```diff
@@ -95,14 +95,8 @@ func (collector *GenericCollector) Name() string {
 
 //Returns collected metrics and the next collection time of the collector
 func (collector *GenericCollector) Collect() (time.Time, []v1.Metric, error) {
-	minNextColTime := collector.configFile.MetricsConfig[0].PollingFrequency
-	for _, metricConfig := range collector.configFile.MetricsConfig {
-		if metricConfig.PollingFrequency < minNextColTime {
-			minNextColTime = metricConfig.PollingFrequency
-		}
-	}
 	currentTime := time.Now()
-	nextCollectionTime := currentTime.Add(time.Duration(minNextColTime * time.Second))
+	nextCollectionTime := currentTime.Add(time.Duration(collector.info.minPollingFrequency * time.Second))
```

Comment:
In line 68: It looks like we don't check if the user provided a polling frequency for any metric config. We need to set in a default if json returns zero and ignore any metricConfig.PollingFrequency which is zero.